packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -229,6 +232,10 @@ class AndroidSdkVersion implements Comparable<AndroidSdkVersion> {
   String get apksignerPath => getBuildToolsPath('apksigner');
 
   List<String> validateSdkWellFormed({bool requireApkSigner = true}) {
+    if (buildToolsVersion.major < minimumAndroidSdkVersion) {
```

Comment:
I thin you are right that there are four artifacts of the Android SDK but I don't think it matters for this CL. Here is a breakdown of how things work today, what this CL does, and what we might want to do in the future:

Today:
Flutter looks at the platform and build tools versions and selects a version of build tools that has the same major version of the platform. The result of this is that flutter guarantees platformVersion.major == buildToolsVersion.major.

Flutter does not look at platform tools or SDK tools today. Maybe it should, I don't know.

This CL:
Detects if the platform/build tool major versions (they are always the same for Flutter right now) are >= 25.

Future:
In the future we may want to validate version numbers of each of these artifacts individually and may eventually need to build a blacklist / whitelist around version combinations.